components/com_fabrik/models/list.php

```diff
@@ -2320,12 +2320,23 @@ public function _addLink($data, &$elementModel, $row, $repeatCounter = 0)
 		$rowId = $this->getSlug($row);
 		$isAjax = $this->isAjaxLinks() ? '1' : '0';
 		$isCustom = $customLink !== '';
+		$isIcon = FALSE;
+		
+		/* Check if the data component is html, if it is an icon lnk and image parse it out */
+		if (class_exists('DOMDocument')) {
+			$html = new DOMDocument;
+			$html->loadHTML($data);
+			$a = $html->getElementsByTagName('a')->item(0);
+			$img = $html->getElementsByTagName('img')->item(0);
+			if (isset($html) && isset($a) && isset($img))
+				$isIcon = TRUE;
+		}
 
 		$paths = array(
 			COM_FABRIK_FRONTEND . '/views/list/tmpl/' . $this->getTmpl() . '/layouts/element/' . $elementModel->getFullName(true, false)
 		);
 
-		$layout                  = $this->getLayout('element.fabrik-element-custom-link', $paths);
+		$layout = $isIcon ? $this->getLayout('element.fabrik-element-custom-icon-link', $paths) : $this->getLayout('element.fabrik-element-custom-link', $paths);
```

Comment:
Avoid long lines, or PHP Storm will perpetually bitch at me about it.  So only use inline condition for short stuff.  Put this in a normal if/else.